oak/server/rust/oak_runtime/tests/integration_test.rs

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::thread::{park, spawn};
+
+mod common {
+    use oak_runtime::runtime::{Handle, Runtime};
+
+    use hyper::{Client, Uri};
+    use log::info;
+    use maplit::hashmap;
+    use oak_abi::OakStatus;
+    use oak_runtime::{config, metrics};
+    use std::sync::Arc;
+    use wat::parse_str;
+
+    const PORT: u16 = 9876;
+
+    pub fn start_runtime() -> Result<(Arc<Runtime>, Handle), OakStatus> {
+        let wat = r#"
+        (module
+            (type (;0;) (func (param i64) (result i32)))
+            (func $oak_main (type 0)
+              i32.const 42)
+            (memory (;0;) 18)
+            (export "memory" (memory 0))
+            (export "oak_main" (func $oak_main)))
+        "#;
+        let binary = parse_str(wat).expect("Could not parse wat module.");
+
+        // Create a runtime with one node
+        let cfg = oak_runtime::application_configuration(
+            hashmap![
+                "node".to_string() => binary,
+            ],
+            "lumberjack",
+            "node",
+            "main",
+        );
+
+        info!("Starting the runtime with one nodes.");
+        config::configure_and_run(cfg)
+    }
+
+    pub fn start_metrics_server() {
+        info!("Starting the metrics server");
+
+        let mut tokio_runtime =
+            tokio::runtime::Runtime::new().expect("Couldn't create Tokio runtime");
+        let result = tokio_runtime.block_on(metrics::serve_metrics(PORT));
+
+        info!("Exiting metrics server node thread {:?}", result);
+    }
+
+    pub async fn read_metrics() -> Result<String, hyper::error::Error> {
+        info!("Reading the metrics.");
+        let client = Client::new();
+
+        let uri = format!("http://localhost:{}", &PORT)
+            .parse::<Uri>()
+            .expect("Could not parse URI.");
+        match client.get(uri).await {
+            Err(e) => Err(e),
+            Ok(res) => {
+                info!("status: {}", res.status());
+
+                // Concatenate the body stream into a single buffer...
+                match hyper::body::to_bytes(res).await {
+                    Err(e) => Err(e),
+                    Ok(buf) => Ok(format!("{:?}", buf)),
+                }
+            }
+        }
+    }
+}
+
+#[test]
+fn test_metrics_gives_the_correct_number_of_nodes() {
+    simple_logger::init().unwrap();
```

Comment:
Is there any advantage in using `simple_logger` vs `env_logger`? I am not familiar with the former.

Reply:
Hmm, we seem to use both in different parts of the codebase:
 - `simple_logger`: used by `oak_glue`, `running_average`, `hello_world`, `private_set_intersection`, `aggregator/module`, `translator`, `abitest`
 - `env_logger`: used by `oak_loader` ,`aggregator/backend`, `split_grpc/{client,proxy}`

Maybe we should settle on one consistently?